apollo-router/src/plugins/telemetry/apollo_exporter.rs

```diff
@@ -140,80 +139,131 @@ impl ApolloExporter {
+            if let Err(e) = self.submit_report(std::mem::take(&mut report)).await {
+                tracing::error!("failed to submit Apollo report: {}", e)
+            }
         });
-        Ok(ApolloExporter { tx })
+        Sender::Apollo(tx)
     }
 
-    pub(crate) fn provider(&self) -> Sender {
-        Sender::Spaceport(self.tx.clone())
-    }
-
-    async fn send_report(
-        pool: &Pool<ReporterManager>,
-        apollo_key: &str,
-        header: &ReportHeader,
-        report: Report,
-    ) {
-        if report.operation_count == 0 && report.traces_per_query.is_empty() {
-            return;
+    pub(crate) async fn submit_report(&self, report: Report) -> Result<(), ApolloExportError> {
+        if report.operation_count == 0 {
+            return Ok(());
         }
+        tracing::debug!("submitting report: {:?}", report);
+        // Protobuf encode message
+        let mut content = BytesMut::new();
+        let mut report = report.into_report(self.header.clone());
+        prost::Message::encode(&report, &mut content)
+            .map_err(|e| ApolloExportError::ClientError(e.to_string()))?;
+        // Create a gzip encoder
+        let mut encoder = GzEncoder::new(Vec::new(), Compression::default());
+        // Write our content to our encoder
+        encoder
+            .write_all(&content)
+            .map_err(|e| ApolloExportError::ClientError(e.to_string()))?;
+        // Finish encoding and retrieve content
+        let compressed_content = encoder
+            .finish()
+            .map_err(|e| ApolloExportError::ClientError(e.to_string()))?;
+        let mut backoff = Duration::from_millis(0);
 
-        match pool.get().await {
-            Ok(mut reporter) => {
-                let report = report.into_report(header.clone());
-                match reporter
-                    .submit(crate::spaceport::ReporterRequest {
-                        apollo_key: apollo_key.to_string(),
-                        report: Some(report),
-                    })
-                    .await
-                {
-                    Ok(_) => {}
-                    Err(e) => {
-                        tracing::warn!("failed to submit stats to spaceport: {}", e);
-                    }
-                };
-            }
-            Err(err) => {
-                tracing::warn!(
-                    "stats discarded as unable to get connection to spaceport: {}",
-                    err
-                );
-            }
-        };
-    }
-}
-
-#[derive(Debug)]
-pub(crate) struct ReporterManager {
-    endpoint: Url,
-}
+        let req = self
+            .client
+            .post(self.endpoint.clone())
+            .body(compressed_content)
+            .header("X-Api-Key", self.apollo_key.clone())
+            .header("Content-Encoding", "gzip")
+            .header(CONTENT_TYPE, "application/protobuf")
+            .header("Accept", "application/json")
```

Comment:
Use constant from http crate for header name

Reply:
[6a1155b](https://github.com/apollographql/router/pull/1970/commits/6a1155b6b7fed93e3b20279fa35ce5b3f92a15ee)